keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,531 @@
+  convince all implementations to support that CRD, this KEP would become moot.
+  That approach would have substantial technical challenges.
+  
+- A general-purpose mechanism for publicly exposing cluster configuration.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Tools that configure a local registry should also apply a ConfigMap that communicates
+"RegistryStatus".
+
+The ConfigMap specifies everything a tool might need to know about how to
+interact with the local registry.
+
+Any tool that pushes an image to a registry should be able to read this
+ConfigMap, and decide to push to the local registry instead.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Alice is setting up a development environment on Kubernetes.
+
+Alice creates a local Kind cluster. She follows the Kind-specific instructions
+for setting up a local registry during cluster creation.
+
+Alice will have some tool she interacts with that builds images and pushes them
+to a registry the cluster can access. That tool might be an IDE like VSCode,
+an infra-as-code toolchain like Pulumi, or a multi-service dev environment like Tilt.
+
+On startup, the tool should connect to the cluster and read the registry ConfigMap.
+
+If the ConfigMap specifies a `help` URL, the tool may prompt the user to set up
+a registry for faster development.
+
+If the ConfigMap specifies a registry location, the tool may automatically
+adjust image pushes to push to the specified registry.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+- Users may see the ConfigMap and draw mistaken conclusions about how to
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- There is no automatic mechanism for keeping the RegistryStatus up-to-date
```

Comment:
this is not necessarily true?
(though it seems perhaps somewhat unreasonable to run a controller just for this)

Reply:
I clarified this a bit - more that this KEP doesn't specify one.